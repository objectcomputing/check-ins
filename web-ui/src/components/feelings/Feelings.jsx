import React from "react";
import PropTypes from "prop-types";

const propTypes = {
  message: PropTypes.string,
  onSelect: PropTypes.func,
};
const displayName = "Feelings";

const Feelings = ({ onSelect, message }) => {
  const inputs = [
    ["Terrible", "fa fa-frown-o fa-3x"],
    ["Bad"],
    ["Okay", "fa fa-meh-o fa-3x"],
    ["Good"],
    ["Great", "fa fa-smile-o fa-3x"],
  ];
  const onChange = (e) => {
    onSelect(e.target.value);
  };

  return (
    <div>
      <h4>{message}</h4>
      <div style={{ display: "flex" }}>
        {inputs.map(([text, icon], i) => (
          <div
            key={`feelings-${i}`}
            style={{
              margin: "10px",
              display: icon === undefined ? "flex" : "",
              alignItems: "flex-end",
            }}
          >
            <div>
              <i className={icon} aria-hidden="true"></i>
            </div>
            <input
              id={`feelings-input-${i}`}
              type="radio"
              name="feeling"
              onClick={onChange}
              value={text}
            />
            {text}
          </div>
        ))}
      </div>
    </div>
  );
};

Feelings.propTypes = propTypes;
Feelings.displayName = displayName;

export default Feelings;
