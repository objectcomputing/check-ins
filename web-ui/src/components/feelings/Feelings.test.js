import React from "react";
import Feelings from "./Feelings";
import { mount } from "enzyme";

it("renders correctly", () => {
  snapshot(<Feelings />);
});

const message = "le test";

it("renders message correctly", () => {
  snapshot(<Feelings message={message} />);
});

it("renders onSelect correctly", () => {
  const onSelect = jest.fn();
  const wrapper = mount(<Feelings onSelect={onSelect} />);
  const find = wrapper.find("#feelings-input-0");
  find.at(0).simulate("click");
  expect(onSelect.mock.calls.length).toBe(1);
});
