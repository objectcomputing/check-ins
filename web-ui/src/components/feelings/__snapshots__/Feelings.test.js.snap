// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h4 />
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "",
          "margin": "10px",
        }
      }
    >
      <div>
        <i
          aria-hidden="true"
          className="fa fa-frown-o fa-3x"
        />
      </div>
      <input
        id="feelings-input-0"
        name="feeling"
        onClick={[Function]}
        type="radio"
        value="Terrible"
      />
      Terrible
    </div>
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "flex",
          "margin": "10px",
        }
      }
    >
      <div>
        <i
          aria-hidden="true"
        />
      </div>
      <input
        id="feelings-input-1"
        name="feeling"
        onClick={[Function]}
        type="radio"
        value="Bad"
      />
      Bad
    </div>
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "",
          "margin": "10px",
        }
      }
    >
      <div>
        <i
          aria-hidden="true"
          className="fa fa-meh-o fa-3x"
        />
      </div>
      <input
        id="feelings-input-2"
        name="feeling"
        onClick={[Function]}
        type="radio"
        value="Okay"
      />
      Okay
    </div>
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "flex",
          "margin": "10px",
        }
      }
    >
      <div>
        <i
          aria-hidden="true"
        />
      </div>
      <input
        id="feelings-input-3"
        name="feeling"
        onClick={[Function]}
        type="radio"
        value="Good"
      />
      Good
    </div>
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "",
          "margin": "10px",
        }
      }
    >
      <div>
        <i
          aria-hidden="true"
          className="fa fa-smile-o fa-3x"
        />
      </div>
      <input
        id="feelings-input-4"
        name="feeling"
        onClick={[Function]}
        type="radio"
        value="Great"
      />
      Great
    </div>
  </div>
</div>
`;

exports[`renders message correctly 1`] = `
<div>
  <h4>
    le test
  </h4>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "",
          "margin": "10px",
        }
      }
    >
      <div>
        <i
          aria-hidden="true"
          className="fa fa-frown-o fa-3x"
        />
      </div>
      <input
        id="feelings-input-0"
        name="feeling"
        onClick={[Function]}
        type="radio"
        value="Terrible"
      />
      Terrible
    </div>
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "flex",
          "margin": "10px",
        }
      }
    >
      <div>
        <i
          aria-hidden="true"
        />
      </div>
      <input
        id="feelings-input-1"
        name="feeling"
        onClick={[Function]}
        type="radio"
        value="Bad"
      />
      Bad
    </div>
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "",
          "margin": "10px",
        }
      }
    >
      <div>
        <i
          aria-hidden="true"
          className="fa fa-meh-o fa-3x"
        />
      </div>
      <input
        id="feelings-input-2"
        name="feeling"
        onClick={[Function]}
        type="radio"
        value="Okay"
      />
      Okay
    </div>
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "flex",
          "margin": "10px",
        }
      }
    >
      <div>
        <i
          aria-hidden="true"
        />
      </div>
      <input
        id="feelings-input-3"
        name="feeling"
        onClick={[Function]}
        type="radio"
        value="Good"
      />
      Good
    </div>
    <div
      style={
        Object {
          "alignItems": "flex-end",
          "display": "",
          "margin": "10px",
        }
      }
    >
      <div>
        <i
          aria-hidden="true"
          className="fa fa-smile-o fa-3x"
        />
      </div>
      <input
        id="feelings-input-4"
        name="feeling"
        onClick={[Function]}
        type="radio"
        value="Great"
      />
      Great
    </div>
  </div>
</div>
`;
